src/glob.rs

```diff
@@ -42,14 +53,14 @@ impl FilePatterns {
       .unwrap_or(true)
   }
 
-  /// Creates a collection of `FilePatterns` by base where the containing patterns
+  /// Creates a collection of `FilePatterns` where the containing patterns
   /// are only the ones applicable to the base.
   ///
   /// The order these are returned in is the order that the directory traversal
   /// should occur in.
-  pub fn split_by_base(&self) -> Vec<(PathBuf, Self)> {
+  pub fn split_by_base(&self) -> Vec<Self> {
     let Some(include) = &self.include else {
-      return Vec::new();
+      return vec![self.clone()];
```

Comment:
Bug fix is here. Previously we were just swallowing this scenario.

I experimented with having `split_by_base` return `Option<Vec<Self>>` in this scenario, but it became really complicated because then a base needed to be passed around with the `FilePatterns`, so I tried creating a `FilePatternsWithBase`, but then it just turned out I basically always needed this, so I just stuck this into `FilePatterns`. Not super ideal, but not terrible.